### PR TITLE
make the created tender appear on buyer's tenderlist - Aisha Athman Lali

### PR DESCRIPTION
**Pull request by aishaathmanlali**
…ing a tender

## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: server/api.js [hunk 1]
```diff
@@ -273,6 +275,10 @@ router.post("/tender", async (req, res) => {
 		newErrors.push("Please select at least one skill.");
 	}
 
+	if (!buyerId) {
+		newErrors.push("Buyer ID is required.");
+	}
+
```
Comment (soonick) [inline server/api.js]: This is unnecessary, buyerId will always be present, otherwise they can't access the endpoint

File: server/api.js [hunk 2]
```diff
@@ -284,8 +290,8 @@ router.post("/tender", async (req, res) => {
 		await client.query("BEGIN");
 
 		const insertTenderQuery = `
-		INSERT INTO tender (title, announcement_date, deadline, description, closing_date, status)
-		VALUES ($1, $2, $3, $4, $5, $6) RETURNING id
+		INSERT INTO tender (title, announcement_date, deadline, description, closing_date, status,  buyer_id, creation_date, last_update, no_of_bids_received)
+		VALUES ($1, $2, $3, $4, $5, $6, $7, $8, CURRENT_DATE, now(), $9) RETURNING id
```
Comment (soonick) [inline server/api.js]: Why do you use `CURRENT_DATE` for one date and `now()` for the other one?

**Review by soonick — CHANGES_REQUESTED** [review]

File: server/api.js (hunk 2 above)
Comment (aishaathmanlali) [reply server/api.js]: Because for the last_update I wanted both date and time while for creation date only the date thats why I used the current_date

Comment (soonick) [reply server/api.js]: I hadn't noticed they are two different types. We should probably have used the same type for both, but it's too late for that. You can ignore this comment.

**Review by soonick — APPROVED** [review]

**Pull request merged**